### PR TITLE
MongoDB Support

### PR DESCRIPTION
**Pull request by agubler**

### CODE REVIEW

File: lib.js [hunk 1]
```diff
@@ -28,32 +33,45 @@ module.exports = class ContribCat {
 		return results;
 	}
 
-	_fetchPullRequests(url, children) {
-		children = children || [];
-		return get(url).spread((response, body) => {
+	_fetchPullRequests(url) {
+		var cutOffDate = this.cutOffDate;
+		return get(url).spread(function (response, body) {
```
Comment (matt-gadd) [inline lib.js]: can we put the arrows back in here?

Comment (agubler) [inline lib.js]: yup

File: index.js [hunk 2]
```diff
@@ -1,6 +1,13 @@
 "use strict";
 var config = require("./config");
+var mongoose = require('mongoose');
 var ContribCat = require("./lib");
 var contribCat = new ContribCat(config);
-contribCat.run().then(contribCat.runReporters.bind(contribCat));
 
+mongoose.connect("mongodb://localhost/contribcat");
```
Comment (matt-gadd) [inline index.js:7]: lets make this a config variable? and all refs to mongoose out of index and server and into lib - given everything else in lib depends on it anyway.

File: lib.js [hunk 3]
```diff
@@ -28,36 +33,49 @@ module.exports = class ContribCat {
...
-				return this._fetchPullRequests(links.next.url, repo, children);
-			}
-			return children;
+
+			var promises = [];
+			items.forEach((item) => {
+				promises.push(PullRequest.createAsync(item).then(function() {}, function() {}));
+			});
+
+			return Promise.all(promises).then(function() {
```
Comment (matt-gadd) [inline lib.js:50]: can use an arrow here

File: lib.js [hunk 4]
```diff
@@ -28,36 +33,49 @@ module.exports = class ContribCat {
...
 			});
-			Array.prototype.push.apply(children, items);
-			if (links && links.next && items.length === body.length) {
-				return this._fetchPullRequests(links.next.url, repo, children);
-			}
-			return children;
+
+			var promises = [];
+			items.forEach((item) => {
+				promises.push(PullRequest.createAsync(item).then(function() {}, function() {}));
```
Comment (matt-gadd) [inline lib.js:47]: can we use `Promise.reflect` here?

File: README.md [hunk 5]
```diff
@@ -1,3 +1,3 @@
-# github-user-contrib
+# contrib-cat
```
Comment (matt-gadd) [inline README.md:1]: :smile:

File: lib.js [hunk 6]
```diff
@@ -28,36 +33,49 @@ module.exports = class ContribCat {
...
 				return moment(item.created_at).isAfter(this.cutOffDate);
 			});
-			Array.prototype.push.apply(children, items);
-			if (links && links.next && items.length === body.length) {
-				return this._fetchPullRequests(links.next.url, repo, children);
-			}
-			return children;
+
+			var promises = [];
+			items.forEach((item) => {
```
Comment (matt-gadd) [inline lib.js:46]: I think we can succinctly write this along the lines of:

``` javascript
Promise.map(items, function(item) {
    return PullRequest.createAsync(item).reflect();
}).then(function() {
    if (links && links.next && items.length === body.length) {
        return this._fetchPullRequests(links.next.url, repo);
    }
});
```

**Comment by matt-gadd** [comment]
@agubler your call with this, we take it as is and we completely refactor afterwards - or we try and clean some of this up in this pr? it's all a bit too nested now imo...

**Pull request merged**